### DISCUSSION TITLE
UIS-RNN base diarization pipeline

### DISCUSSION THREAD

**Discussion by malarinv** [pyannote/pyannote-audio#792] [Q&A]
The current diarization model/pipeline is based on clustering and it seems to have its own problems (`All samples have mutually equal similarities. Returning arbitrary cluster center(s)` etc.) .  

Is there any plan to implement [UIS-RNN](https://github.com/google/uis-rnn) based (Fully supervised diarization) model pipeline?

**Comment by hbredin** [comment]
I have no such plan. Why don't you use the linked UIS-RNN repo directly since it has been implemented there already?

**Reply by malarinv** [reply]
I am using that already, but the api interface/pipeline is much cleaner with pyannote.
